src/services/exportWallet.service.ts

```diff
@@ -3,36 +3,35 @@ import crypto from 'node:crypto';
 import { JWTIssuer, Scope } from '@zelusio/auth-lib';
 import JWKService from './jwk/jwk.service';
 import { secrets } from './secrets.service';
-
-export interface IAuthWalletEncryptedBody {
-  jwt_user: string;
-}
+import { HEADER_NAME } from '../utilities/requestId';
 
 export interface IAuthWalletEncryptedResponse {
   encrypted_wallet: string;
 }
 
-export async function getEncryptedWallet(authToken: string, subject: string): Promise<string> {
+export async function getEncryptedWallet(authToken: string, subject: string, requestId: string): Promise<string> {
   const keyServiceInstance = await JWKService.getProviderInstance();
 
   const jwtIssuer = new JWTIssuer(secrets.NSS_ISSUER, await keyServiceInstance.getDefaultSigningKey());
   const nssToken = await jwtIssuer.issueAccessToken(subject, [], [Scope.InvisibleWalletExport]);
-  const walletEncrypted = await getEncryptedWalletForUser(authToken, nssToken);
+  const walletEncrypted = await getEncryptedWalletForUser(authToken, nssToken, requestId);
 
   return walletEncrypted.encrypted_wallet;
 }
 
 export async function getEncryptedWalletForUser(
   authToken: string,
-  nssToken: string
+  nssToken: string,
+  requestId: string
 ): Promise<IAuthWalletEncryptedResponse> {
   const result = await axios.post(
     `${secrets.AUTH_API_URL}/wallet/encrypted`,
     {},
     {
       headers: {
         Authorization: `Bearer ${authToken}`,
-        'nss-auth': `Bearer ${nssToken}`
+        'nss-auth': `Bearer ${nssToken}`,
```

Comment:
Using non-standard HTTP headers is fine, we just need to make sure to use them correctly. Non-standard headers should be prefixed with `X-` e.g. `X-Nss-Auth`